src/Core/src/Platform/iOS/LabelExtensions.cs

```diff
@@ -23,5 +24,31 @@ public static void UpdateTextColor(this UILabel nativeLabel, ILabel label)
 				nativeLabel.TextColor = textColor.ToNative(textColor);
 			}
 		}
+
+		public static void UpdateTextDecorations(this UILabel nativeLabel, ILabel label)
+		{
+			if (nativeLabel.AttributedText != null && !(nativeLabel.AttributedText?.Length > 0))
+				return;
+
+			var textDecorations = label?.TextDecorations;
+
+			var newAttributedText = nativeLabel.AttributedText != null ? new NSMutableAttributedString(nativeLabel.AttributedText) : new NSMutableAttributedString(label?.Text ?? string.Empty);
```

Comment:
How does this fit with the check above? At the top, if there is not attrText, then bail. Bu here we check again.